### PR TITLE
feat: settings screen

### PR DESCRIPTION
**Pull request by tormodAase**
This PR implements Settings Screen into mad-core

### CODE REVIEW

File: apps/chronicles/mad.config.ts [hunk 1]
```diff
@@ -17,4 +20,11 @@ export const config: MadConfig = {
         title: "Chronicles",
         logo: Logo,
     },
+    serviceNow: {
+        whatever: "",
```
Comment (HavardNot) [inline apps/chronicles/mad.config.ts:24]: This looks like a placeholder? Maybe add a todo if you are unable to implement this at the moment?

File: apps/chronicles/navigation/SettingsScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                {
+                    name: "switch",
+                    title: "Test",
+                    onChange: () => undefined,
+                    isActive: true,
+                    iconName: "abacus",
+                },
+            ],
+        },
+    ];
```
Comment (HavardNot) [inline apps/chronicles/navigation/SettingsScreen.tsx:30]: Is this a template? Should maybe clarify it, e.g. rename to "templateConfig" or something

File: packages/core/src/components/screens/settings/types.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          title: string;
+          isActive: boolean;
+          onChange: (isActive: boolean) => void;
+          switchSize?: "small" | "normal";
+          disabled?: boolean;
+          description?: string;
+          color?: SwitchColor;
+          iconName: IconName;
+      }
+    | { name: "custom"; title: string; iconName: IconName; param: unknown };
```
Comment (HavardNot) [inline packages/core/src/components/screens/settings/types.tsx:31]: These types could maybe be unified with the type of the component it describes. E.g. "switch" should use the type of the Switch component props, "button" the Button component props, and so on.

File: packages/core/src/types.ts [hunk 4]
```diff
@@ -59,6 +59,20 @@ export type MadConfig = {
...
+         * Endpoints used by the app
+         */
+        endpoints: string[];
+        /**
+         * Build number of the app.
+         */
+        buildNumber: string;
+    };
+    serviceNow?: {
+        //TODO
```
Comment (HavardNot) [inline packages/core/src/types.ts:73]: Nice todo bro 😎

File: packages/core/tsconfig.json [hunk 5]
```diff
@@ -8,7 +8,8 @@
         "noPropertyAccessFromIndexSignature": true,
         "noImplicitReturns": true,
         "noFallthroughCasesInSwitch": true,
-        "exactOptionalPropertyTypes": true,
+        //"exactOptionalPropertyTypes": true,
```
Comment (HavardNot) [inline packages/core/tsconfig.json:11]: Is this temporarily commented out or can it be removed?

**Review by HavardNot — CHANGES_REQUESTED** [review]
Looks great! I have a few QoL requested changes, but other than that we should be good to go! 🎉

File: apps/chronicles/mad.config.ts (hunk 1 above)
Comment (HavardNot) [reply apps/chronicles/mad.config.ts:24]: Edit: I noticed you added a todo to the type declaration so looks like youre covered 👍

File: packages/core/tsconfig.json (hunk 5 above)
Comment (tormodAase) [reply packages/core/tsconfig.json:11]: Removed, hard to use mad-components with that setting enabled, when mad-components does not have it enabled

File: packages/core/src/components/screens/settings/types.tsx (hunk 3 above)
Comment (tormodAase) [reply packages/core/src/components/screens/settings/types.tsx]: Fixed

File: apps/chronicles/navigation/SettingsScreen.tsx (hunk 2 above)
Comment (tormodAase) [reply apps/chronicles/navigation/SettingsScreen.tsx]: Fixed

**Review by HavardNot — APPROVED** [review]
Looks hella fresh 🧔🏻‍♂️

**Pull request merged**